aspnet/fundamentals/static-files.rst

```diff
@@ -1,8 +1,336 @@
+
+Let's take the following directory/file hierarchy example to illustrate:
+
+- wwwroot
+
+  - css
+  - images
+
+    - test.image
+
+  - ...
+
+Using this hierarchy, you could enable static file serving and directory browsing with the following:
+
+.. code-block:: c#
+  :emphasize-lines: 5-6
+
+  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+  {
+    ...
+    // Serve static files and allow directory browsing.
+    app.UseDirectoryBrowser();
+    app.UseStaticFiles();
+
+If the user browses to ``http://<yourApp>/images``, a directory listing will be displayed by the browser that includes the ``test.image`` file. However, if the user clicks on that file, they will see a 404 error - even though the file obviously exists. In order to allow the serving of unknown file types, you could set the ``StaticFileOptions.ServeUnknownFileTypes`` property to ``true``:
+
+.. code-block:: c#
+  :emphasize-lines: 5-6
+
+  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+  {
+    ...
+    // Serve static files, including unknown file types, and allow directory browsing.
+    app.UseDirectoryBrowser();
+    app.UseStaticFiles(new StaticFileOptions { ServeUnknownFileTypes = true });
+
+Now, ASP.NET will attempt to serve any file - known and unknown - and it's up to the browser to render the file correctly based on the file's contents.
+
+But, what if you have a folder that contains files whose types are unknown to ASP.NET and you know that they're all a specific type. In that example, you could opt to specify a default content type for file types that ASP.NET doesn't recognize. The following code example illustrates how to specify the default content type for unknown file types is application/x-msdownload (downloadable application). For a list of all MIME types, see this `list <http://www.freeformatter.com/mime-types-list.html>`_.
+
+.. code-block:: c#
+  :emphasize-lines: 5-8
+
+  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+  {
+    ...
+    // Serve static files and allow directory browsing.
+    app.UseDirectoryBrowser();
+    app.UseStaticFiles(new StaticFileOptions
+    {
+      ServeUnknownFileTypes = true,
+      DefaultContentType = "application/x-msdownload"
+    });
+
+At this point, if the user browses to a file whose content type is unknown, the browser will treat it as a downloadable file and offer the option to download the file (or simply download it automatically, depending on the browser).
+
+So far, you've seen how to specify a default content type for any file type that ASP.NET doesn't recognize. However, what if you have a situation where you have several different file types that are unknown to ASP.NET? That's where the ``IContentTypeProvider`` interface comes in.
```

Comment:
Most users should instantiate FileExtensionContentTypeProvider and adding their own types to the list. IContentTypeProvider is for advanced customization.